### PR TITLE
add configuration format guide

### LINKED ISSUE #138: doc: write a guide to convert the configuration between different formats
Write a guide to convert between yaml, json and graphql formats. This can be done using the compose command with `--format` modifier as documented here — https://tailcall.run/docs/guides/cli/#--format

## Content Creation Requirements

To maintain the quality of our content, please adhere to the following guidelines:

### 1. Accuracy in Language

- **Grammar and Spelling:** Ensure your content is free from grammatical and spelling errors.

### 2. Tone and Style

- **Neutral Tone:** Maintain a neutral, non-emotional tone throughout the content.
- **Engaging Style:** Write in a free-flowing and engaging manner, keeping the reader's interest.

### 3. Content Integrity

- **Fact-Checking:** Verify all information. If unsure, use Discord to clarify.
- **Relevance:** Ensure all content is cohesive, to the point, and directly related to the title.

### 4. Originality

- **Avoid Low-Effort Content:** Content should be original and not solely generated by AI tools like ChatGPT.

**PS: Adherence to these guidelines is crucial. Content not meeting these standards may require revision or may not be accepted.**

### PR DESCRIPTION
**Pull request by neo773**
/claim #138
/closes #138

### CODE REVIEW

File: docs/guides/compose.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+title: Compose
+---
+
+The Tailcall CLI's `compose` command facilitates the merging of configuration files and their conversion between different formats using the `--format` option.
+
+It supports `gql` or `graphql`, `yml` or `yaml`, `json`. Default is `json`.
+
+## Converting Formats with `compose --format`
```
Comment (tusharmath) [inline docs/guides/compose.md]: avoid writing monospace in headings. Bad SEO and doesn't look that good when HTML is rendered.

File: docs/guides/compose.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+title: Compose
+---
+
+The Tailcall CLI's `compose` command facilitates the merging of configuration files and their conversion between different formats using the `--format` option.
```
Comment (tusharmath) [inline docs/guides/compose.md]: Intro should talk about the fact that tailcall support three input formats in configurations.

File: docs/guides/compose.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+```bash
+tailcall compose examples/jsonplaceholder.graphql --format json > "examples/jsonplaceholder.json"
+```
+
+Now to convert back to graphql
+
+```bash
+tailcall compose examples/jsonplaceholder.json --format graphql > "examples/jsonplaceholder2.graphql"
+```
```
Comment (tusharmath) [inline docs/guides/compose.md]: We should use tabs to showcase different versions of the configuration using tabs. Check this `configuration.mdx` out.

File: docs/guides/compose.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Converting Formats with `compose --format`
+
+To convert files between different formats, use the following syntax:
+
+```bash
+tailcall compose <input_files> <output_file> --format format_type
+```
+
+## Example Usage
```
Comment (tusharmath) [inline docs/guides/compose.md]: Should link to `operators` as the natural next step to learn more about how to write the configuration.

File: docs/guides/compose.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
+title: Compose
```
Comment (tusharmath) [inline docs/guides/compose.md]:
```suggestion
title: Configuration Format
```

Comment (tusharmath) [reply docs/guides/compose.md]: This guide is about configuration formats and not compose.

File: docs/guides/compose.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+---
```
Comment (tusharmath) [inline docs/guides/compose.md]: Need a note about using `.tailcallrc.graphql` and `.tailcallrc.schema.json` with json and yml.

Comment (tusharmath) [reply docs/guides/compose.md]: @neo773 need to add some context about these files as well. The files help in autocomplete. Would be great to showcase that feature here.

File: docs/guides/compose.mdx [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+}
+```
+</TabItem>
+</Tabs>
+
+### Enhanced Editor Support
+
+To leverage autocomplete and validation for Tailcall configurations, utilize `.tailcallrc.graphql` for GraphQL configurations and `.tailcallrc.schema.json` for JSON and YAML configurations. These files enhance editor support by providing schema definitions, facilitating faster and error-free configuration.
+
+For more detailed information on configuring TailCall for enhanced editor support, refer to tailcall's [init command](/docs/guides/cli.md#init).
```
Comment (tusharmath) [inline docs/guides/compose.mdx]: @neo773 I think we should expand a little bit more on how to use these files. For example how can I setup `.tailcallrc.graphql` with `.graphqlrc.yml` and use `.tailcallrc.schema.json` with a json file. A step by step guide would be useful here.

**Comment by neo773** [comment]
@tusharmath 
Added changes

**Review by tusharmath — APPROVED** [review]

**Pull request merged**